src/type/definition.js

```diff
@@ -184,20 +303,36 @@ export function assertCompositeType(type: ?GraphQLType): GraphQLCompositeType {
  */
 export type GraphQLAbstractType = GraphQLInterfaceType | GraphQLUnionType;
 
-export function isAbstractType(type: ?GraphQLType): boolean %checks {
-  return (
-    type instanceof GraphQLInterfaceType || type instanceof GraphQLUnionType
-  );
+export function isAbstractType(type: mixed): boolean %checks {
+  return isInterfaceType(type) || isUnionType(type);
 }
 
-export function assertAbstractType(type: ?GraphQLType): GraphQLAbstractType {
+export function assertAbstractType(type: mixed): GraphQLAbstractType {
   invariant(
     isAbstractType(type),
     `Expected ${String(type)} to be a GraphQL abstract type.`,
   );
   return type;
 }
 
+/**
+ * These types wrap and modify other types
+ */
+
+export type GraphQLWrappingType = GraphQLList<any> | GraphQLNonNull<any>;
+
+export function isWrappingType(type: mixed): boolean %checks {
+  return isListType(type) || isNonNullType(type);
+}
+
+export function assertWrappingType(type: mixed): GraphQLWrappingType {
+  invariant(
+    isWrappingType(type),
+    `Expected ${String(type)} to be a GraphQL wrapping type.`,
```

Comment:
@leebyron Since this error messages are intended for developers who just use public API they can unfamiliar with what `wrapping type` means. Maybe it worth to replace it with something like  `to be a type wrapped by either NonNull or List wrapper`?
Similarly, other error messages will benefit from replacing composite terms like `leat type`, `abstract type`, `composite type`, etc.